modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/challenge/questions/ServerChallengeTestCase.java

```diff
@@ -0,0 +1,346 @@
+
+        List<String> actualQuestion = getValueKeyInNestedArray(response.asString(), questionObjectAttribute,
+                questionAttribute);
+        Assert.assertTrue(actualQuestion.contains(question), "Provided question value does not exist");
+
+    }
+
+    /**
+     * The method is used to validate the response which as JSON Array.
+     *
+     * @param jsonArrayStr represent JSON Array.
+     * @param key will give you the value of a particular key of the JSON Array.
+     * @return will return the corresponding values.
+     * @throws JSONException thrown to indicate a problem with the JSON.
+     */
+    private List<String> getValuesByKeyInJSONArray(String jsonArrayStr, String key) throws JSONException {
+        List<String> values = new ArrayList<>();
+        JSONArray jsonArray = new JSONArray(jsonArrayStr);
+        for (int idx = 0; idx < jsonArray.length(); idx++) {
+            JSONObject jsonObj = jsonArray.getJSONObject(idx);
+            values.add(jsonObj.optString(key));
+        }
+        return values;
+    }
+
+    /**
+     * The method is used to validate the response which has JSON nested Array.
+     *
+     * @param jsonArrayStr represent JSON Array.
+     * @param key will give you the value of a particular key of the JSON Array.
+     * @param nestedArrayKey represent Nested JSON Array.
+     * @return will return the corresponding values.
+     * @throws JSONException thrown to indicate a problem with the JSON.
+     */
+    private List<String> getValueKeyInNestedArray(String jsonArrayStr, String key, String nestedArrayKey) throws
+            JSONException {
+
+        List<String> values = new ArrayList<>();
+        JSONArray jsonArray = new JSONArray(jsonArrayStr);
+        for (int idx = 0; idx < jsonArray.length(); idx++) {
+            JSONObject jsonObj = jsonArray.getJSONObject(idx);
+            String questionObject = jsonObj.getString(key);
+            JSONArray nestedArray = new JSONArray(questionObject);
+
+            for (int k = 0; k < nestedArray.length(); k++) {
+                JSONObject nestedobj = nestedArray.getJSONObject(k);
+
+                values.add(nestedobj.optString(nestedArrayKey));
+
+            }
+
+        }
+        return values;
+    }
+
+    /**
+     *The method is used to validate the response which as JSON object.
+     *
+     * @param responseasString string, the response which will return as Json Object
+     * @param setIdKey string, challenge question setId
+     * @param questionIdKey string, challenge question id
+     * @param questionKey String, challenge question
+     * @param localekey String, challenge question locale
+     * @throws JSONException thrown to indicate a problem with the JSON.
+     */
+    private void recurseKeys(String responseasString, String setIdKey, String questionIdKey, String questionKey,
+            String localekey)
+            throws
+            JSONException {
```

Comment:
Can come to a single line. 
Also, add an empty line at the beginning of the method. Check for other places as well.

Reply:
Modified